lib/toolshed/net.ex

```diff
@@ -168,12 +168,20 @@ defmodule Toolshed.Net do
   end
 
   defp resolve_addr(address) do
-    with {:ok, hostent} <- :inet.gethostbyname(to_charlist(address)),
-         hostent(h_addr_list: ip_list) = hostent,
-         first_ip = hd(ip_list) do
-      {:ok, first_ip}
-    else
-      _ -> {:error, "Error resolving #{address}"}
+    address_families = [:inet, :inet6]
+
+    Enum.reduce(address_families, nil, fn
+      _family, {:ok, _hostent} = acc -> acc
+      family, _acc -> :inet.gethostbyname(to_charlist(address), family)
+    end)
+    |> case do
+      {:ok, hostent} ->
+        hostent(h_addr_list: ip_list) = hostent
+        first_ip = hd(ip_list)
+        {:ok, first_ip}
+
+      _ ->
+        {:error, "Error resolving #{address}"}
```

Comment:
How about something like this?

```suggestion
    case gethostbyname(address, :inet) || gethostbyname(address, :inet6) do
      nil -> {:error, "Error resolving #{address}"}
      ip -> {:ok, ip}
    end
  end

  defp gethostbyname(address, family) do
    case :inet.gethostbyname(to_charlist(address), family) do
      {:ok, hostent} ->
        hostent(h_addr_list: ip_list) = hostent
        hd(ip_list)

      _ ->
        nil
    end
```

Reply:
I tried that option when writing the PR and it was my second choice because it has a disadvantage to consider. By using `address_families = [:inet, :inet6]` in the version I committed it makes it easier to add an option like `ping("google.com", family: :inet6)` which would force only one address family to be used, in this case to ping Google's IPv6 public address (the current implementation will only ping that domain via IPv4). That refactor becomes simpler by changing the address families in the list that's iterated over rather than modifying conditional logic. I do get that at first glance the `||` code seems easier to read, and that holds true if there's no desire for a `family: :inet6` option.

Just thought I'd throw that out there if you like the idea of a `family: :inet6` option. I'm ultimately ok with either implementation of the code.

Reply:
I don't mind the `family: inet6` option or perhaps since this is a helper function, forgoing the keyword and making `address_families` a positional argument. I'd try using `Enum.find_values` since that has the lazy eval semantics of `||`. Anyway, that seems like we could look at it in a follow-up PR.

Could you accept the code that I provided assuming that it doesn't have a bug?

Reply:
Sure! I can change the code to use your approach, test it, and push it.

Reply:
Updated in 814aff8